homeassistant/components/select/__init__.py

```diff
@@ -72,6 +72,8 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 class SelectEntityDescription(EntityDescription):
     """A class that describes select entities."""
 
+    options: list[str] = UNDEFINED  # type: ignore[assignment]
```

Comment:
This feels way to hacky for my taste.

Reply:
It's convenience. Same reason why we ignore type for `hass` in `Entity`.

Reply:
What we do in `hass` is a very exceptional case.
In this case, `options` cannot be `None`, so we don't need to use sentinel values. 

```suggestion
    options: list[str] | None = None
```